src/Bucket/HTTPRequest.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace HelloFresh\Stats\Bucket;
+
+
+use HelloFresh\Stats\HTTPMetricAlterCallback;
+use Psr\Http\Message\RequestInterface;
+
+class HTTPRequest extends Plain
+{
+    /** @var RequestInterface */
+    protected $request;
+    /** @var HTTPMetricAlterCallback */
+    protected $httpMetricAlterCallback;
+
+    /**
+     * HTTPRequest constructor.
+     *
+     * @param string $section
+     * @param RequestInterface $request
+     * @param bool $success
+     * @param HTTPMetricAlterCallback|null $callback
+     */
+    public function __construct($section, RequestInterface $request, $success, HTTPMetricAlterCallback $callback = null)
+    {
+        $this->request = $request;
+        $this->httpMetricAlterCallback = $callback;
```

Comment:
Why do we need to store `request` and `httpMetricAlterCallback`? Why not simply add them as args to `buildMetricOperation` and store that output?